[docs][iOS][location] Added collapsible section that tries to explain iOS location permissions (#35453)
expo/expo: docs/pages/versions/unversioned/sdk/location.mdx

- On Android, you must request both foreground and background permissions.
- On iOS, it must be granted with the `Always` option using [`requestBackgroundPermissionsAsync`](#locationrequestbackgroundpermissionsasync).

<Collapsible summary="Expo and iOS permissions">
iOS permissions are divided into the two categories `When In Use` and `Always` and maps to Expo's foreground and background location permissions requested via:

> - [`requestForegroundPermissionsAsync`](#locationrequestforegroundpermissionsasync) → `When In Use`

Reviewer: ```suggestion
- [`requestForegroundPermissionsAsync`](#locationrequestforegroundpermissionsasync) maps to `When In Use`
```

Let's avoid using a symbol.

> - [`requestBackgroundPermissionsAsync`](#locationrequestbackgroundpermissionsasync) → `Always`

Reviewer: ```suggestion
- [`requestBackgroundPermissionsAsync`](#locationrequestbackgroundpermissionsasync) maps to `Always`
```

> > **NOTE:** When requesting `When In Use` authorization, the user can grant **temporary access** by selecting `Allow Once` in the system permission dialog. This authorization will be valid **only for the current app session** and is automatically revoked when the app is closed.

Reviewer: ```suggestion
> **Note:** When requesting `When In Use` authorization, the user can grant **temporary access** by selecting `Allow Once` in the system permission dialog. This authorization will be valid **only for the current app session** and is automatically revoked when the app is closed.
```

**Detecting "Allow Once" versus "Allow While Using the App"**

Unfortunately, **iOS does not provide a way to detect whether the user selected `Allow Once` or `Allow While Using the App`**. Both responses result in `When In Use` authorization.

> If the user selected `Allow Once` and you subsequently call [`requestBackgroundPermissionsAsync`](#locationrequestbackgroundpermissionsasync) in the same session, the system **will not show another prompt**. Instead the request will **silently fail** and the returned background permission status will be **denied**.

Reviewer: ```suggestion
If the user selected `Allow Once` and you subsequently call [`requestBackgroundPermissionsAsync`](#locationrequestbackgroundpermissionsasync) in the same session, the system **will not show another prompt**. Instead, the request will **silently fail**, and the returned background permission status will be **denied**.
```

**Handling "Allow Once" scenarios**

> If you suspect the user selected `Allow Once` and need to request background permissions, they must **manually enable background location** in the Settings app:

Reviewer: ```suggestion
If you suspect the user selected `Allow Once` and needs to request background permissions, they must **manually enable background location** in the Settings app. You can use `Linking` to open the Settings app within your app:
```

- Grammatical typo
- Adding some context about the code snippet below does or how it can be used from a developer's app, just to avoid confusion

```js
import { Linking } from 'react-native';

function openSettings() {
  Linking.openURL('app-settings:');
}
```

> **Incremental Permission Requests**

Reviewer: ```suggestion
**Incremental permission requests**
```

> It is possible to request foreground location access first, then ask for background location access later. This can improve the user experience by requesting permissions only when necessary.

Reviewer: ```suggestion
It is possible to request foreground location access first and then ask for background location access later. This can improve the user experience by requesting permissions only when necessary.
```

> **Requesting Background Permissions Directly**

Reviewer: ```suggestion
**Requesting Background Permissions directly**
```

We are switching from sentence case (first two sections) and title case (in last two sections). Let's stick to using one pattern for improved readability
